tst/unit/test_output_utils.cpp

```diff
@@ -0,0 +1,281 @@
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include <cstdio>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "basic_types.hpp"
+#include "globals.hpp"
+#include "interface/mesh_data.hpp"
+#include "interface/meshblock_data.hpp"
+#include "interface/metadata.hpp"
+#include "interface/state_descriptor.hpp"
+#include "interface/variable_pack.hpp"
+#include "kokkos_abstraction.hpp"
+#include "mesh/domain.hpp"
+#include "mesh/meshblock.hpp"
+#include "outputs/output_utils.hpp"
+
+using parthenon::BlockList_t;
+using parthenon::DevExecSpace;
+using parthenon::IndexDomain;
+using parthenon::IndexShape;
+using parthenon::loop_pattern_mdrange_tag;
+using parthenon::MeshBlock;
+using parthenon::MeshBlockData;
+using parthenon::MeshData;
+using parthenon::Metadata;
+using parthenon::PackIndexMap;
+using parthenon::par_for;
+using parthenon::Real;
+using parthenon::StateDescriptor;
+
+using namespace parthenon::OutputUtils;
+
+TEST_CASE("The VarInfo object produces appropriate ranges", "[VarInfo][OutputUtils]") {
+  GIVEN("A MeshBlock with some vars on it") {
+    constexpr int NG = 2;
+    constexpr int NSIDE = 16;
+    constexpr int NDIM = 3;
+    constexpr int NFULL = (NSIDE + 2 * NG);
+
+    constexpr auto interior = parthenon::IndexDomain::interior;
+    constexpr auto entire = parthenon::IndexDomain::entire;
+
+    // JMM: This needs to be reset to 0 when we're done, because other
+    // tests assume it's unset, thus zero-initialized.
+    parthenon::Globals::nghost = NG;
+
+    auto pkg = std::make_shared<StateDescriptor>("Test package");
+
+    const std::string scalar_cell = "scalar_cell";
+    Metadata m({Metadata::Cell, Metadata::Independent});
+    pkg->AddField(scalar_cell, m);
+
+    const std::string tensor_cell = "tensor_cell";
+    m = Metadata({Metadata::Cell, Metadata::Independent}, std::vector<int>{3, 4});
+    pkg->AddField(tensor_cell, m);
+
+    const std::string tensor_none = "tensor_none";
+    m = Metadata({Metadata::None, Metadata::Independent}, std::vector<int>{3, 4});
+    pkg->AddField(tensor_none, m);
+
+    // four-vector-valued var with a vector at each face
+    const std::string vector_face = "four_vector_face";
+    m = Metadata({Metadata::Face, Metadata::Independent}, std::vector<int>{4});
+    pkg->AddField(vector_face, m);
+
+    // vector-valued var with a single value at each edge
+    const std::string scalar_edge = "scalar_edge";
+    m = Metadata({Metadata::Edge, Metadata::Independent});
+    pkg->AddField(scalar_edge, m);
+
+    const std::vector<std::string> var_names = {scalar_cell, tensor_cell, tensor_none,
+                                                vector_face, scalar_edge};
+
+    auto pmb = std::make_shared<MeshBlock>(NSIDE, NDIM);
```

Comment:
Given the bug @adamdempsey90 found wrt to indices, would it make sense to test non-cubic blocks? (though I'm not suggesting to change it here as I'm not sure how much effort that would be and if it's worth it in the context of this PR).

Reply:
This seems like it would be a little bit of a pain in the unit tests here... but I agree. See issue #1041 .